src/components/compose.jsx

```diff
@@ -600,15 +600,31 @@ function Compose({
     const handleItems = (e) => {
       const { items } = e.clipboardData || e.dataTransfer;
       const files = [];
+      const unsupportedFiles = [];
       for (let i = 0; i < items.length; i++) {
         const item = items[i];
         if (item.kind === 'file') {
           const file = item.getAsFile();
-          if (file && supportedMimeTypes.includes(file.type)) {
+          if (!file) {
+            alert(t`Unable to attach file.`);
+            return;
+          }
+          else if (supportedMimeTypes !== undefined && !supportedMimeTypes.includes(file.type)) {
+            unsupportedFiles.push(file);
+          }
+          else {
             files.push(file);
           }
         }
       }
+      if (unsupportedFiles.length > 0) {
+        alert(
+          plural(unsupportedFiles.length, {
+            one: `File ${unsupportedFiles[0]} is not supported by your instance.`,
+            other: `Files ${unsupportedFiles.join(", ")} are not supported by your instance.`,
```

Comment:
1. " by your instance" is not needed, in my opinion.
2. the `join` is better done with [`toLocaleString`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toLocaleString) or [`Intl.ListFormat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/ListFormat).

Reply:
1. Wouldn't the user think that the issue is with Phanpy instead of the server they're using, though?

Reply:
> 1. Wouldn't the user think that the issue is with Phanpy instead of the server they're using, though?

The confusion is _already_ happening with all other features, not just on Phanpy, but also on other 3rd-party mobile apps from my observation 😂. E.g. app settings vs instance settings is the more common confusion, yet at the same time we also don't want users to think too much. The confusion is more apparent as most folks migrate from other centralised only-one-app services.